lib/statesman/adapters/active_record.rb

```diff
@@ -70,7 +70,7 @@ def create_transition(from, to, metadata)
 
         transition = transitions_for_parent.build(transition_attributes)
 
-        ::ActiveRecord::Base.transaction do
+        @parent_model.with_lock do
```

Comment:
Actually this is probably the best option. Any locks acquired will be held until the end of the transaction anyway (I think).